### PR TITLE
Add code for Maximum Weight Independent Set of Path Graph DP problem in C++

### PR DESCRIPTION
**Pull request by GuptaSamridhi**
**Changes:**

Add code for Maximum Weight Independent Set of Path Graph DP problem in C++

### CODE REVIEW

File: code/dynamic_programming/src/maximum_weight_independent_set_of_path_graph/maximum_weight_independent_set_of_path_graph.cpp [hunk 1]
```diff
@@ -0,0 +1,37 @@
+#include <iostream>
+using namespace std;
+
+int max(int a, int b) {
```
Comment (YashovardhanSiramdas) [inline code/dynamic_programming/src/maximum_weight_independent_set_of_path_graph/maximum_weight_independent_set_of_path_graph.cpp]: ```max``` function is already available in iostream header file.

File: code/dynamic_programming/src/maximum_weight_independent_set_of_path_graph/maximum_weight_independent_set_of_path_graph.cpp [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    return a>b?a:b;
+}
+
+int main() {
```
Comment (YashovardhanSiramdas) [inline code/dynamic_programming/src/maximum_weight_independent_set_of_path_graph/maximum_weight_independent_set_of_path_graph.cpp]: Braces should be in their own line

File: code/dynamic_programming/src/maximum_weight_independent_set_of_path_graph/maximum_weight_independent_set_of_path_graph.cpp [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    int n;
+    cout << "Enter the number of vertices of the path graph : ";
+    cin >> n;
+    int g[n];
+    cout << "Enter the vertices of the graph : ";
+    for (int i=0; i<n; i++) {
```
Comment (YashovardhanSiramdas) [inline code/dynamic_programming/src/maximum_weight_independent_set_of_path_graph/maximum_weight_independent_set_of_path_graph.cpp]: No need of braces if loop contains single line of code

File: code/dynamic_programming/src/maximum_weight_independent_set_of_path_graph/maximum_weight_independent_set_of_path_graph.cpp [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        cin >> g[i];
+    }
+    int sol[n+1];
+    sol[0] = 0;
+    sol[1] = g[0];
+    for (int i=2; i<=n; i++) {
```
Comment (YashovardhanSiramdas) [inline code/dynamic_programming/src/maximum_weight_independent_set_of_path_graph/maximum_weight_independent_set_of_path_graph.cpp]: Same as above

File: code/dynamic_programming/src/maximum_weight_independent_set_of_path_graph/maximum_weight_independent_set_of_path_graph.cpp [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+        sol[i] = max(sol[i-1], sol[i-2]+g[i-1]);
+    }
+    cout << "The maximum weighted independent set sum is : " << sol[n] << endl;
+    int i = n;
+    cout << "The selected verices are : ";
+    while (i >= 1) {
```
Comment (YashovardhanSiramdas) [inline code/dynamic_programming/src/maximum_weight_independent_set_of_path_graph/maximum_weight_independent_set_of_path_graph.cpp]: Braces in their own line

File: code/dynamic_programming/src/maximum_weight_independent_set_of_path_graph/maximum_weight_independent_set_of_path_graph.cpp [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+	if (sol[i-1] >= sol[i-2]+g[i-1]) {
```
Comment (YashovardhanSiramdas) [inline code/dynamic_programming/src/maximum_weight_independent_set_of_path_graph/maximum_weight_independent_set_of_path_graph.cpp]: No need of braces

File: code/dynamic_programming/src/maximum_weight_independent_set_of_path_graph/maximum_weight_independent_set_of_path_graph.cpp [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+	    i--;
+	}
+	else {
```
Comment (YashovardhanSiramdas) [inline code/dynamic_programming/src/maximum_weight_independent_set_of_path_graph/maximum_weight_independent_set_of_path_graph.cpp]: Braces in their own line

**Comment by GuptaSamridhi** [comment]
I have addressed the comments in my next commit.

**Comment by AdiChat** [comment]
Looks great 😍 

Hope you are enjoying your journey 🚆  with **OpenGenus**

**Pull request merged**